include/grpcpp/impl/codegen/server_callback.h

```diff
@@ -81,9 +156,8 @@ class CallbackUnaryHandler : public MethodHandler {
 
     if (status.ok()) {
       // Call the actual function handler and expect the user to call finish
-      CatchingCallback(std::move(func_), param.server_context,
-                       controller->request(), controller->response(),
-                       controller);
+      CatchingCallback(func_, param.server_context, controller->request(),
```

Comment:
Why do we lose the move() here?

Reply:
We should have never had the move. That leaves an object in undefined state and that wasn't valid for this class since the func_ is per-method and not per-rpc. However, maybe we can change this to take a reference argument.

Reply:
I want to hold off on evaluating whether CatchingCallback should use a move ref or const ref until we know whether this matters (as I think it'll just get inlined and not copied anyway).

Reply:
I'm making this decision a todo